### ISSUE TITLE
about interleave the input

### ISSUE THREAD

**Issue by YUE-FAN** [valencebond/FixMatch_pytorch#1]
Hi, thanks for the nice work.

I have a small question about the `interleave` code. I think the batch normalization will still have a biased estimation after interleaving `imgs` because `imgs` is fed into the model as a whole, so the `running_mean` and `running_var` will be the same as the ones computed without `interleave`.

According to the original paper, I guess we need to call `model.eval()` when evaluating pseudo labels for the strongly augmented unlabeled images.

**Comment by YUE-FAN** [comment]
Sorry, I was wrong about the argument about the `running_mean` and `running_var`. We should use the concatenated input to update the `running_mean` and `running_var` because it will give a more accurate estimation.

But still, I think the `interleave` does not make a difference in single GPU training here :)

**Comment by valencebond** [comment]
Hi, @YUE-FAN , for the interleave operation , i agree with you and  i just refer to the origin tensorflow implementation. 

 For the pseudo label, i can not find the accurate operation in the original paper  Fixmatch and i am not sure whether to using model.eval(). I refer to the MixMatch "We do not propagate gradients through computing the guessed labels,". And in orginal [tf implementation](https://github.com/google-research/fixmatch/blob/08d9b83d7cc87e853e6afc5a86b12aacff56cdea/fixmatch.py#L113), the operation is  just to stop the gradients propagation.

thanks for your advices, please correct me if i am wrong

**Comment by YUE-FAN** [comment]
Thanks for your reply. Your code is correct. We should not call `model.eval()`.

My previous concern about `model.eval()` was related to the `running_mean` and `running_var` in batch normalizations. I thought we should not allow strongly augmented unlabeled images to contribute to the `running_mean` and `running_var` (calling `model.eval()` will switch them into eval mode). 
But actually it is better to incorporate strongly augmented unlabeled images for BN layer because a larger batch size will give a more accurate estimation to the `running_mean` and `running_var` :)
